pub.go

```diff
@@ -77,6 +82,30 @@ func (pub *pubSocket) SetOption(name string, value interface{}) error {
 	return pub.sck.SetOption(name, value)
 }
 
+// GetTopics is used to retrieve subscribed topics for a pub socket.
+func (pub *pubSocket) Topics() []string {
+	pub.sck.mu.RLock()
+	t := []string{}
```

Comment:
instead of two for-loops, I'd do it in go.
_ie:_
```go
var (
    keys = make(map[string]struct{})
    topics []string
)
for _, con := range pub.sck.conns {
    con.mu.RLock()
    for topic := range con.topics {
        if _, dup := keys[topic]; dup {
            continue
        }
        keys[topic] = struct{}{}
        topics = append(topics, topic)
    }
    con.mu.RUnlock()
}
return topics
```